modules/c/nitf/source/ImageIO.c

```diff
@@ -9482,14 +9480,16 @@ nitf_ImageIO_12PixelComStart(nitf_CompressionControl *object,
   icntl->offset = offset;
   icntl->blockMask = blockMask;
   icntl->padMask = padMask;
-  icntl->buffer = NULL;
   icntl->written = 0;
 
 /* Allocate compressed block buffer */
 
-  icntl->buffer = (nitf_Uint8 *) NITF_MALLOC(icntl->blockSizeCompressed);
-  if(icntl->buffer == NULL)
-    return(NITF_FAILURE);
+  if (!icntl->buffer)
```

Comment:
Same difference but change this check to `icntl->buffer == NULL` just for code legibility